src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -264,13 +275,17 @@ public async Task<HttpResponseMessage> HandleRequestAsync(HttpRequestMessage req
                 {
                     basePath = this.localHttpListener.InternalRpcUri.AbsolutePath;
                 }
-                else if (this.durableTaskOptions.NotificationUrl != null)
-                {
-                    basePath = this.durableTaskOptions.NotificationUrl.AbsolutePath;
-                }
                 else
                 {
-                    throw new InvalidOperationException($"Don't know how to handle request to {request.RequestUri}.");
+                    Uri webhookUrl = this.GetWebhookUri();
+                    if (webhookUrl != null)
+                    {
+                        basePath = webhookUrl.AbsolutePath;
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Don't know how to handle request to {request.RequestUri}.");

```

Comment:
Could we provide a more descriptive error? For example, isn't the problem here that the webhook isn't available?
Additionally, since GetWebhookUri throws an exception, would we ever reach this else statement?

Reply:
You are right that it is no longer necessary, since GetWebhookUri already throws by this point.